### PR TITLE
Add unit test to Ratchet

### PR DESCRIPTION
**Pull request by Johann-S**
Hi,

We talked about to add unit test to Ratchet (https://github.com/twbs/ratchet/issues/723).
This is why I made this PR.

I chose to use Jasmine. 
It's the beginning I'll add more unit tests if it's ok.

/CC @cvrebert

### CODE REVIEW

**Comment by cvrebert** [comment]
CC: @XhmikosR @connors

File: .gitignore [hunk 1]
```diff
@@ -37,3 +37,4 @@ nbproject
 # Folders to ignore
 node_modules
 bower_components
+.grunt
```
Comment (XhmikosR) [inline .gitignore]: What does it create this folder?

**Comment by XhmikosR** [comment]
I think it's very nice work :+1: Sure, it doesn't cover all cases but it's a beginning.

I'm not sure if we should use mocha, qunit or jasmine, something else though.

File: js/tests/commonSpec.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+describe('Common', function () {
+  it('RATCHET namespace is define', function () {
```
Comment (XhmikosR) [inline js/tests/commonSpec.js]: defined

File: js/tests/commonSpec.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    expect(typeof RATCHET !== 'undefined').toBe(true);
+  });
+
+  it('window.CustomEvent exist', function () {
```
Comment (XhmikosR) [inline js/tests/commonSpec.js]: exists

File: js/tests/commonSpec.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    expect(typeof window.CustomEvent !== 'undefined').toBe(true);
+  });
+
+  it('RATCHET.getBrowserCapabilities return an object', function () {
```
Comment (XhmikosR) [inline js/tests/commonSpec.js]: returns

File: js/tests/commonSpec.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    var result = RATCHET.getBrowserCapabilities;
+    expect(typeof result === 'object').toBe(true);
+  });
+
+  it('RATCHET.getTransitionEnd return string', function () {
```
Comment (XhmikosR) [inline js/tests/commonSpec.js]: returns

File: js/tests/SpecRunner.html [hunk 6]
```diff
@@ -0,0 +1,65 @@
+<!doctype html>
+<html>
+<head>
+  <meta charset="utf-8">
+  <title>Jasmine Spec Runner</title>
+  <link rel="shortcut icon" type="image/png" href=".grunt/grunt-contrib-jasmine/jasmine_favicon.png">
+
+  <link rel="stylesheet" type="text/css" href="../../.grunt/grunt-contrib-jasmine/jasmine.css">
```
Comment (XhmikosR) [inline js/tests/SpecRunner.html]: Maybe remove type from `link`s since it's the default.

**Comment by cvrebert** [comment]
Anyways, I'm not part of Team Ratchet and this PR is nontrivial, so I'm unsubscribing from this.

File: .gitignore (hunk 1 above)
Comment (tqdjr1) [inline .gitignore]: Wow I'm not to I sure what's up here but first time and already I'm getting response I like it but want trying to ask if my work made any sense but thanks  you'll see me again tjankyoi

Comment (Johann-S) [inline .gitignore]: This folder is created by Jasmine and have Jasmine dependencies.

File: js/tests/commonSpec.js (hunk 4 above)
Comment (Johann-S) [inline js/tests/commonSpec.js]: Sorry for my bad English

File: js/tests/SpecRunner.html (hunk 6 above)
Comment (Johann-S) [inline js/tests/SpecRunner.html]: SpecRunner.html is generated by Jasmine each time we run ``grunt jasmine``  I'm not sure I can specify to change how the CSS files will be include.

Comment (XhmikosR) [inline js/tests/SpecRunner.html]: No worries it's nitpicking anyway :)

**Comment by Johann-S** [comment]
Is it a problem if I include JQuery to emulate touch event like this : 
``var event = $.Event( "touchstart", { pageX:200, pageY:200 } );``

Because I'm working on creating a unit test to check if the slide event is trigger and I cannot
create a ``CustomEvent`` with an array of touches.
I think it would be easier to work with touch event emulation.

**Comment by Johann-S** [comment]
I added a few unit tests for Modal and Slider.

And I made a little script which allow you to fake touch event 
https://github.com/Johann-S/TouchFaker

I used my script to send a fake touchstart in the slider unit test.
Because it's not possible ta create a CustomEvent and add an array of touches this is why
I made this script.

**Comment by XhmikosR** [comment]
For some reason I didn't get the last CC. I'll try to have a look in the next days and merge this. Should be good as a starting point.

**Comment by Johann-S** [comment]
Thank you :smile:

**Comment by Johann-S** [comment]
Sorry @XhmikosR but... PING :smile:

**Comment by XhmikosR** [comment]
@Johann-S: this needs to be rebased.

**Comment by Johann-S** [comment]
Done @XhmikosR :wink:

**Comment by XhmikosR** [comment]
@Johann-S: is it possible to show only the tick or X in test output instead of duplicating the test message? Now it's like `Slider should fire slide event...✓ Slider should fire slide event`

**Comment by Johann-S** [comment]
I can change the display to this by using ``display: 'short'`` in jasmine configuration

image

**Comment by XhmikosR** [comment]
@Johann-S: yes better use that.

**Comment by Johann-S** [comment]
Do you want I squash my commits @XhmikosR ?

**Comment by XhmikosR** [comment]
Yes, please.

**Comment by Johann-S** [comment]
Done :+1:

**Comment by XhmikosR** [comment]
@Johann-S: What is the reason you didn't use https://github.com/hammerjs/touchemulator?

**Comment by Johann-S** [comment]
I fork touchemulator because touchemulator only allow you to fake touch events on mouse events and what I wanted to do it was to trigger fake touch event on unit tests so that's why I create touchfaker

**Comment by Johann-S** [comment]
Thank you for your time and feedbacks @XhmikosR

**Pull request merged**